### PR TITLE
Add base precompile

### PR DESCRIPTION
**Pull request by deblanco**
## Description

Implemented frontier precompiled contracts using the typing of Moonbeam.
Added a native token erc-20 precompile with mint & burn support.

## Motivation

Solve token issuance

## Impact on the project

mod `runtime/src/precompile.rs`
mod `runtime/src/stability_config.rs`
created `precompiles`

### CODE REVIEW

File: runtime/src/stability_config.rs [hunk 1]
```diff
@@ -17,3 +20,7 @@ pub const MAXIMUM_BLOCK_WEIGHT: Weight = WEIGHT_PER_MILLIS
     .div(COMPUTATION_BLOCK_TIME_RATIO.1); // 1_333_333_333_333
 
 pub const MAXIMUM_BLOCK_LENGTH: u32 = u32::MAX;
+
+parameter_types! {
```
Comment (GabrielMartinezRodriguez) [inline runtime/src/stability_config.rs:24]: In my opinion would be a better practice if in this file we define the param `DEFAULT_OWNER`, an then in the `runtime/src/lib.rs` declare de parameter type using the param `DEFAULT_OWNER`:

`runtime/src/stability_config.rs`:

```rs
pub const DEFAULT_OWNER:  &str = "0xa58482131a8d67725e996af72D91A849AcC0F4A1";
```

`runtime/src/lib.rs`:
```rs
parameter_types! {
  pub DefaultOwner : H160 = H160::from_str(DEFAULT_OWNER).expect("invalid address");
}
```

**Comment by GabrielMartinezRodriguez** [comment]
Good job @clostao.  Give your opinion about my suggested changes

File: runtime/src/stability_config.rs (hunk 1 above)
Comment (clostao) [reply runtime/src/stability_config.rs:24]: I think that extracting the value from the type is a great approach, however I would extract the H160 not only the string.

Comment (GabrielMartinezRodriguez) [reply runtime/src/stability_config.rs:24]: Ok

Comment (clostao) [reply runtime/src/stability_config.rs:24]: I'll proceed with my approach

**Review by GabrielMartinezRodriguez — APPROVED** [review]

**Review by clostao — APPROVED** [review]

**Pull request merged**